test/dynamo/test_misc.py

```diff
@@ -10117,6 +10117,30 @@ def fn(x, y):
 
         self.assertEqual(actual, expected)
 
+    def test_frozen_dataclass_field_without_init(self):
+        @dataclasses.dataclass(frozen=True)
+        class TestDataClass:
+            x: torch.Tensor
+            n: int = dataclasses.field(init=False)
+
+            def __post_init__(self):
+                object.__setattr__(self, "n", self.x.shape[0])
+
+        @allow_in_graph
+        def inner_fn(dc):
+            return dc.x + dc.n
+
+        def fn(x):
+            dc = TestDataClass(x)
+            return inner_fn(dc)
+
+        fn_opt = torch.compile(fullgraph=True)(fn)
```

Comment:
backend="eager", we don't want to run inductor when we don't need to

Reply:
Updated.